src/t_stream.c

```diff
@@ -1338,6 +1345,132 @@ robj *createObjectFromStreamID(streamID *id) {
                         id->ms,id->seq));
 }
 
+/* Get the first or last entry ID from a non-empty stream. */
+void streamGetTipID(stream *s, streamID *id, int first) {
```

Comment:
minor: there's already an `streamGetEdgeID` (which does the same, but includes tombstone entries)
maybe somehow unite them and add `int include_tombstones`?

Reply:
That's definitely good idea - initially I used it but the tombs were problematic. OTOH, maybe just keep a "first_id" at the stream level?

Reply:
Rethinking about streamGetEdgeID, that would mean bringing into it most of the logic streamIteratorGetID, so maybe I'll just stick with that for now.

Reply:
FYIs, regardless of the interface/s question, this also adds a 'first_id' field to the stream struct.

Reply:
The two were merged with a boolean flag.